x-pack/plugins/ingest_manager/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -416,13 +410,53 @@ const updateExistingIndex = async ({
     // if update fails, rollover data stream
   } catch (err) {
     try {
+      // get the data_stream values to compose datastream name
+      const searchDataStreamFieldsResponse = await callCluster('search', {
+        index: indexTemplate.index_patterns[0],
+        body: {
+          size: 1,
+          query: {
+            bool: {
+              must: [
+                {
+                  exists: {
+                    field: 'data_stream.type',
+                  },
+                },
+                {
+                  exists: {
+                    field: 'data_stream.dataset',
+                  },
+                },
+                {
+                  exists: {
+                    field: 'data_stream.namespace',
+                  },
+                },
+              ],
+            },
+          },
+        },
+      });
+      if (searchDataStreamFieldsResponse.hits.total.value === 0)
+        throw new Error('data_stream fields are missing from datastream indices');
+      const {
+        dataset,
+        namespace,
+        type,
+      }: {
+        dataset: string;
+        namespace: string;
+        type: DataType;
+      } = searchDataStreamFieldsResponse.hits.hits[0]._source.data_stream;
+      const dataStreamName = `${type}-${dataset}-${namespace}`;
```

Comment:
nit: do we want to check that `type` is a valid `DataType`? I suppose if they ingested data initially with an invalid data type then we should just continue doing that 🤷

Reply:
Yea, i'm not sure this is the place to be validating that and whether we'd have to throw an error and stop the upgrade because of that.